backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/PatientRegistrationController.java

```diff
@@ -119,6 +130,26 @@ public ResponseEntity<PatientRegistrationResponse> getPatientRegistration(
 		}
 	}
 
+	/**
+	 * The Payee number submitted in the request must match the Payee Number mapped to the current user in the BCSC to Payee Number mappings.
+	 * 
+	 * @param patientRegistrationRequest
+	 * @throws HNWebException
+	 */
+	private void validatePayeeNumberMapping(PatientRegistrationRequest patientRegistrationRequest)
+			throws HNWebException {
+		UserInfo userInfo = SecurityUtil.loadUserInfo();
+		BcscPayeeMapping bcscPayeeMapping = bcscPayeeMappingService.find(userInfo.getUserId());
+		if (bcscPayeeMapping == null) {
+			logger.error("No Payee Number mapping was found for the current user");
+			throw new HNWebException(ExceptionType.PAYEE_MAPPING_NOT_FOUND);
```

Comment:
Removed throwing of HNWeb exceptions. Made the validate method return type void and throws ResponseStatusException from the method so that I could keep custom message for each error.